site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/repository/FhirRepository.java

```diff
@@ -18,18 +19,44 @@
      */
     String saveOrganization(Organization organization);
 
+    /**
+     * Find Organization By name from the FHIR store
+     *
+     * @param name of the organization to search.
+     * @return Organization searched by name or none if not found
+     */
+    Optional<Organization> findOrganizationByName(String name);
+
     /**
      * Creates a new ResearchStudy resource.
-     * @param researchStudy
+     *
+     * @param researchStudy study to create
      * @return ResearchStudy
      */
     String saveResearchStudy(ResearchStudy researchStudy);
 
     /**
      * Return the list of all organizations. Note this may include organizations that are
      * not {uk.ac.ox.ndph.mts.site_service.model.Site}s - caller must filter.
+     *
      * @return all organization instances in the store, might be empty, not null
      */
-    Collection<Organization> findOrganizations();
+    List<Organization> findOrganizations();
+
+    /**
+     * Return an organization by ID, or Optional.none() if not found
+     *
+     * @param id organization ID to search for
+     * @return optional with the organization or none if not found
+     */
+    Optional<Organization> findOrganizationById(String id);
+
+    /**
+     * Return an organizations with given parent ID, or Optional.none() if not found
+     *
+     * @param id parent organization ID to search for, null is allowed (will return orgs with no parent)
+     * @return collection of organizations with the given parent, might be empty
+     */
+    List<Organization> findOrganizationsByPartOf(String id);
```

Comment:
findOrganizationsByParent(String id) maybe better as we are searing for organizations using an id that represents its parent? I think findOrganizationsByPartOf gives the impression that you are searching for an organization with a part of its name?

Reply:
This is a FHIR service though, so should use FHIR terminology - the search is on the Organization.partOf attribute. The business SiteService would use findByParent() (this endpoint does not exist yet).